### PR TITLE
Addition of Database Operations

### PR DESCRIPTION
**Pull request by balajisandiparthi2002**
Connected Java application with the database. Inserted the data into audit table.

### CODE REVIEW

File: src/main/java/org/LogAggregatorTool/operations/TimestampSorter.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            new SimpleDateFormat(LogAggregatorConstants.NEW_DATE_FORMAT)
+    );
+
+    private static Date parseDate(String dateString) {
+        for (SimpleDateFormat dateFormat : DATE_FORMATS) {
+            try {
+                return dateFormat.parse(dateString);
+            } catch (ParseException parseException) {
+                //parseException iterates for every mismatch.So, not printing anything.
+            }
```
Comment (vinodkadam22) [inline src/main/java/org/LogAggregatorTool/operations/TimestampSorter.java]: Do not comment line 27, Try handling it

File: SQL Scripts/Table Creation.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+	record_id int not null auto_increment primary key,
+    path_to_the_folder varchar(100),
+    number_of_files int,
+    names_of_files varchar(1000),
+    date_time varchar(50),
+    result varchar(50),
+    output_file_name varchar(50),
+    error_message varchar(100)
+);
+
```
Comment (vinodkadam22) [inline SQL Scripts/Table Creation.sql]: Remove extra lines

File: src/main/java/org/LogAggregatorTool/constants/DatabaseConstants.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package org.LogAggregatorTool.constants;
+
+public class DatabaseConstants {
+    public static final String AUDIT_INSERT_QUERY = "insert into audit(path_to_the_folder, number_of_files, names_of_files, date_time, result, output_file_name, error_message) values(?,?,?,?,?,?,?)";
+    public static final String DATABASE_URL = "jdbc:mysql://localhost:3306/poc1";
```
Comment (vinodkadam22) [inline src/main/java/org/LogAggregatorTool/constants/DatabaseConstants.java:5]: Duplicate quires in 2 files?

File: src/main/java/org/LogAggregatorTool/dto/LogAggregatorAuditData.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
+package org.LogAggregatorTool.dto;
+
+public class LogAggregatorAuditData {
+    private String pathToTheFolder;
+    private int numberOfLogFiles;
+    private String namesOfLogFiles;
+    private String dateTime;
```
Comment (vinodkadam22) [inline src/main/java/org/LogAggregatorTool/dto/LogAggregatorAuditData.java:7]: Why cant you use Lombok?

File: src/main/java/org/LogAggregatorTool/jdbc/LogAggregatorAuditDataEntry.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            auditPreparedStatement.setString(1, logAggregatorAuditData.getPathToTheFolder());
+            auditPreparedStatement.setInt(2, logAggregatorAuditData.getNumberOfLogFiles());
+            auditPreparedStatement.setString(3, logAggregatorAuditData.getNamesOfLogFiles());
+            auditPreparedStatement.setString(4, logAggregatorAuditData.getDateTime());
+            auditPreparedStatement.setString(5, logAggregatorAuditData.getResult());
+            auditPreparedStatement.setString(6, logAggregatorAuditData.getOutputFileName());
+            auditPreparedStatement.setString(7, logAggregatorAuditData.getErrorMessage());
+            auditPreparedStatement.executeUpdate();
+        } catch (SQLException sqlException) {
+            System.out.println(LogAggregatorConstants.DATABASE_ERROR + sqlException.getMessage());
```
Comment (vinodkadam22) [inline src/main/java/org/LogAggregatorTool/jdbc/LogAggregatorAuditDataEntry.java:31]: Even Add printStacktrace

File: src/main/java/org/LogAggregatorTool/operations/LogFilesProcessor.java
Comment (vinodkadam22) [inline src/main/java/org/LogAggregatorTool/operations/LogFilesProcessor.java:1]: Remove unwanted Lines

File: src/main/java/org/LogAggregatorTool/dto/LogAggregatorAuditData.java (hunk 4 above)
Comment (balajisandiparthi2002) [reply src/main/java/org/LogAggregatorTool/dto/LogAggregatorAuditData.java:7]: Lombok dependency is not working in my system.

File: src/main/java/org/LogAggregatorTool/operations/TimestampSorter.java (hunk 1 above)
Comment (balajisandiparthi2002) [reply src/main/java/org/LogAggregatorTool/operations/TimestampSorter.java]: Resolved

File: SQL Scripts/Table Creation.sql (hunk 2 above)
Comment (balajisandiparthi2002) [reply SQL Scripts/Table Creation.sql]: Resolved

File: src/main/java/org/LogAggregatorTool/constants/DatabaseConstants.java (hunk 3 above)
Comment (balajisandiparthi2002) [reply src/main/java/org/LogAggregatorTool/constants/DatabaseConstants.java:5]: Resolved

File: SQL Scripts/Insert Data.sql [hunk 6]
```diff
@@ -0,0 +1,2 @@
+use POC1;
+insert into audit(path_to_the_folder, number_of_files, names_of_files, date_time, result, output_file_name, error_message) values(?,?,?,?,?,?,?);
```
Comment (divya-sathyan) [inline SQL Scripts/Insert Data.sql]: Add extra line

**Review by ashutosh-2609 — APPROVED** [review]

**Review by divya-sathyan — APPROVED** [review]

**Review by vinodkadam22 — APPROVED** [review]

**Pull request merged**